parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java

```diff
@@ -74,19 +78,28 @@ public static WriterVersion fromString(String name) {
   private final int dictionaryPageSizeThreshold;
   private final WriterVersion writerVersion;
   private final boolean enableDictionary;
+  // TODO - Should this even hold an allocator? this isn't really a property
+  private ByteBufferAllocator allocator;
 
   public ParquetProperties(int dictPageSize, WriterVersion writerVersion, boolean enableDict) {
+    this(dictPageSize, writerVersion, enableDict, new HeapByteBufferAllocator());
+  }
+
+  public ParquetProperties(int dictPageSize, WriterVersion writerVersion, boolean enableDict, ByteBufferAllocator allocator) {
```

Comment:
let's not multiply public constructors. Just make one private constructor with all fields.
You can make a pattern like this: (not a builder but similar idea)
```
public ParquetProperties withAllocator(ByteBufferAllocator allocator) {
  return new ParquetProperties(this.dictPageSize, this.writerVersion, this.enableDict, allocator);
}
```
Probably we should have one parameterLess public constructor and deprecate the others.